File: BraintreeModels/src/main/java/com/braintreepayments/api/InvalidAuthorization.java
package com.braintreepayments.api;

class InvalidAuthorization extends Authorization {

    private final String errorMessage;

    InvalidAuthorization(String rawValue, String errorMessage) {
        super(rawValue);
        this.errorMessage = errorMessage;
    }

    @Override
    String getConfigUrl() {
        return null;
    }

    @Override
    String getBearer() {
        return null;
    }

    String getErrorMessage() {
        return errorMessage;
    }
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/InvalidAuthorization.kt
package com.braintreepayments.api

import androidx.annotation.RestrictTo

/**
 * @suppress
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
class InvalidAuthorization(rawValue: String, val errorMessage: String) :
    Authorization(rawValue) {

    override val configUrl: String? = null
    override val bearer: String? = null
}
